src/traces/parcoords/defaults.js

```diff
@@ -51,29 +51,41 @@ function dimensionsDefaults(traceIn, traceOut) {
             continue;
         }
 
-        coerce('label');
-        coerce('tickvals');
-        coerce('ticktext');
-        coerce('visible');
-        coerce('range');
-        coerce('constraintrange');
-        coerce('values');
+        var userVisible = coerce('visible');
 
         // turn dimensions with no data invisible
-        dimensionOut.visible = dimensionOut.visible && dimensionOut.values.length > 0;
+        var actuallyVisible = userVisible && Lib.isArray(dimensionIn.values) && dimensionIn.values.length > 0;
+
+        if(actuallyVisible) {
```

Comment:
Thanks for the direction, I now handpicked the rest of your commit - please eyeball [the changes](https://github.com/plotly/plotly.js/pull/1256/commits/55b41deb67ce2623ce0f585a53529a31fd2b2da1) to the impacted four jasmine tests, hopefully they are in line with how you wanted it to work. (20ms `setTimeout` is unrelated)